app/components/candidate_interface/continuous_applications/application_submit_component.html.erb

```diff
@@ -14,6 +14,14 @@
       <% if error.type == :incomplete_details %>
         <p class="govuk-body">You need to <%= govuk_link_to('complete your details', candidate_interface_continuous_applications_details_path) %> before you can submit this application.</p>
         <p class="govuk-body">This application will be saved as a draft while you finish your details.</p>
+      <% elsif error.type == :immigration_status %>
```

Comment:
This is getting interesting.

I suppose this is because we can't put markup in the locales files?

I'm looking into this right now.

I'll probably dynamically render a partial based on the `error.type`

Reply:
> I'll probably dynamically render a partial based on the error.type
This sounds a good idea, will do in your branch now that validations are defined and grouped?